lib/regentanz/template_compiler.rb

```diff
@@ -80,6 +70,25 @@ def upload_template(stack_path, template)
       'AWS::StackName',
     ].map!(&:freeze).freeze
 
+    def upload_template(stack_path, template)
+      if @template_url
+        if (captures = @template_url.match(%r{\As3://(?<bucket>[^/]+)/(?<key>.+)\z}))
+          bucket = captures[:bucket]
+          bucket = bucket.sub('${AWS_REGION}', @region)
+          key = captures[:key]
+          key = key.sub('${TEMPLATE_NAME}', File.basename(stack_path))
+          key = key.sub('${TIMESTAMP}', Time.now.to_i.to_s)
```

Comment:
Any particular reason for doing `sub` instead of `gsub`? And any particular reason for restricting some parameters to the bucket and some to the key? Some combinations are obviously not that meaningful, but the code wouldn't really become any more complicated by allowing all variables everywhere (and any number of times)